src/session.go

```diff
@@ -143,6 +170,33 @@ func validateSessionTicket(toa *TraefikOidcAuth, encryptedTicket string) (*sessi
 	return session, claims, nil, nil
 }
 
+func checkSessionExpiresSoon(toa *TraefikOidcAuth, session *session.SessionState) bool {
+	expiresSoon := false
+	if toa.Config.SessionCookie.MaxAge > 0 && !session.ExpiresAt.IsZero() {
+		remainingDuration := time.Until(session.ExpiresAt)
+
+		halfMaxAge := float64(toa.Config.SessionCookie.MaxAge) / 2
+
+		if remainingDuration.Seconds() < halfMaxAge {
+			expiresSoon = true
+			toa.logger.Log(logging.LevelDebug, "The session is halfway through it's expiration. Renewing now...")
+		}
+	}
+
+	if !expiresSoon {
+		pastDuration := time.Since(session.RefreshedAt)
+
+		halfMaxAge := float64(session.TokenExpiresIn) / 2
```

Comment:
Ideally, this `2` could be configurable (maybe as some fraction between `0` and `1`), but a hardcoded value is okay too, and half is good enough for that.